presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -175,15 +175,22 @@ private PlanBuilder appendInPredicateApplyNodes(PlanBuilder subPlan, Set<InPredi
 
     private PlanBuilder appendInPredicateApplyNode(PlanBuilder subPlan, InPredicate inPredicate, boolean correlationAllowed)
     {
+        if (subPlan.canTranslate(inPredicate)) {
+            // given subquery is already appended
+            return subPlan;
```

Comment:
can you add a test which check if there are to `IN` subquery expression only one `ApplyNode` (JoinNode) is added to plan